### PR TITLE
make testing better mimic reality for securitymanager

### PR DESCRIPTION
**Pull request by rmuir**
The two things are not the same today: so in tests we should use the same technique, and install policy etc the same way. But today tests do things differently than bin/elasticsearch, like use a java.security.policy system property. And this hides bugs!

Once its fixed to work the same way, we see the bugs, the java.io.tmpdir sysprop substitution does not work (#10925 this is some chicken/egg problem in the jvm). We don't use this method and instead dynamically generate it just like other paths in environment.

Second, today things like SSL are not going to work at all. That is because we don't have any grants for system permissions for jdk codesources (but when you set a sysprop, like tests were doing, it sucks in default files).

Finally, this has the advantage that now tests too, don't depend on the environment, like policy configurations on the machine that someone may have changed.

### CODE REVIEW

**Comment by s1monw** [comment]
good catch rob! LGTM

File: src/main/resources/org/elasticsearch/bootstrap/security.policy [hunk 1]
```diff
@@ -21,15 +21,36 @@
...
+grant codeBase "file:${java.home}/lib/-" {
+        permission java.security.AllPermission;
+};
+
+// These permissions apply to all shared system extensions
+grant codeBase "file:${java.home}/jre/lib/ext/-" {
+        permission java.security.AllPermission;
+};
+
+// These permissions apply to javac when ${java.home] points at $JAVA_HOME/jre
```
Comment (rjernst) [inline src/main/resources/org/elasticsearch/bootstrap/security.policy]: ] -> } after java.home?

**Comment by rmuir** [comment]
@rjernst @s1monw i pushed a new commit: https://github.com/rmuir/elasticsearch/commit/072b90296aa0cea3b684cce2cba56ef459096bdc

File: src/main/java/org/elasticsearch/bootstrap/Security.java [hunk 2]
```diff
@@ -60,10 +52,13 @@ static void configure(Environment environment) throws Exception {
     }
 
     /** returns dynamic Permissions to configured paths */
-    static Permissions createPermissions(Environment environment) throws IOException {
+    static Permissions createPermissions(Environment environment, boolean addTempDir) throws IOException {
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/bootstrap/Security.java]: can the 2 tests calling this with false just override java.io.tmpdir for the test? then no need for a new boolean?

**Comment by s1monw** [comment]
still LGTM

**Comment by rjernst** [comment]
LGTM, I pushed a change to remove the flag for adding tmp dir permissions.

**Comment by rmuir** [comment]
Thanks for that test improvement @rjernst

**Pull request merged**